### PR TITLE
 Initial support for EKS manifests

### PR DESCRIPTION
**Pull request by falfaro**
Add manifest(s) to support EKS on AWS.

### CODE REVIEW

**Comment by jjo** [comment]
I see some non-EKS related `externaldns.jsonnet` changes for
https://github.com/kubernetes-incubator/external-dns#the-latest-release-v05
(allowing `external-dns` to be aware of the RRs it's managing),
suggest moving this off this PR.

File: manifests/platforms/eks.jsonnet [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Top-level file for AWS EKS
+
+local kube = import '../lib/kube.libsonnet';
```
Comment (jjo) [inline manifests/platforms/eks.jsonnet]: Use double-quotes (as existing entries), also fix several other lines below

File: manifests/platforms/eks.jsonnet [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      spec+: {
+        template+: {
+          spec+: {
+            containers_+: {
+              proxy+: {
+                args_+: {
+                  provider: "oidc",
+                  "oidc-issuer-url": "https://cognito-idp.%s.amazonaws.com/%s" % [
+                    $.config.oauthProxy.aws_region,
+                    $.config.oauthProxy.aws_user_pool_id
```
Comment (jjo) [inline manifests/platforms/eks.jsonnet]: Style: add trailing comma,

File: manifests/platforms/eks.jsonnet [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    ingress+: {
+      host: "prometheus." + $.external_dns_zone_name,
+    },
+  },
+
+  fluentd_es: fluentd_es {
+    es:: $.elasticsearch,
+  },
+
+  elasticsearch: elasticsearch {
```
Comment (jjo) [inline manifests/platforms/eks.jsonnet]: Minor: suggest removing `{\n}` no-op (unless you want/need it for some specific reason).

**Comment by jjo** [comment]
> LGTM in general, some style-ish suggestions

And asking to move-off the `external-dns` changes.

File: manifests/platforms/eks.jsonnet (hunk 1 above)
Comment (falfaro) [reply manifests/platforms/eks.jsonnet]: Funny, this was @anguslees undoing my changes (well, actually the changes by Visual Studio Code). Willdo.

Comment (jjo) [reply manifests/platforms/eks.jsonnet]: Whatever makes sense, as far as it's consistent.
Kinda obvious canonical plug:
https://dev.to/__shadz_/tabs-vs-space-49l5

Comment (jjo) [reply manifests/platforms/eks.jsonnet]: Still seeing those 2 style observations (single+double quotes mix[1], missing ending comma),
@falfaro: I'll wait for your fixes.

Comment (falfaro) [reply manifests/platforms/eks.jsonnet]: Fixed.

**Comment by falfaro** [comment]
PTAL

**Pull request merged**